qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -0,0 +1,421 @@
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Built-in transpiler stage plugins for preset pass managers."""
+
+from qiskit.transpiler.passmanager_config import PassManagerConfig
+from qiskit.transpiler.passmanager import PassManager
+from qiskit.transpiler.exceptions import TranspilerError
+from qiskit.transpiler.passes import BasicSwap
+from qiskit.transpiler.passes import LookaheadSwap
+from qiskit.transpiler.passes import StochasticSwap
+from qiskit.transpiler.passes import SabreSwap
+from qiskit.transpiler.passes import Error
+from qiskit.transpiler.preset_passmanagers import common
+from qiskit.transpiler.preset_passmanagers.plugin import PassManagerStagePlugin
+
+
+class DefaultRoutingPassManager(PassManagerStagePlugin):
+    """Plugin class for default routing stage."""
+
+    def pass_manager(self, pass_manager_config: PassManagerConfig) -> PassManager:
+        """Build routing stage PassManager."""
+        opt_level = pass_manager_config.optimization_level
+        seed_transpiler = pass_manager_config.seed_transpiler
+        target = pass_manager_config.target
+        coupling_map = pass_manager_config.coupling_map
+        backend_properties = pass_manager_config.backend_properties
+        if opt_level == 0:
+            routing_pass = StochasticSwap(coupling_map, trials=20, seed=seed_transpiler)
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map=coupling_map,
+                seed_transpiler=seed_transpiler,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 1:
+            routing_pass = StochasticSwap(coupling_map, trials=20, seed=seed_transpiler)
+            vf2_call_limit = None
+            if (
+                pass_manager_config.layout_method is None
+                and pass_manager_config.initial_layout is None
+            ):
+                vf2_call_limit = int(5e4)  # Set call limit to ~100ms with retworkx 0.10.2
+
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map,
+                vf2_call_limit=vf2_call_limit,
+                backend_properties=backend_properties,
+                seed_transpiler=seed_transpiler,
+                check_trivial=True,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 2:
+            routing_pass = StochasticSwap(coupling_map, trials=20, seed=seed_transpiler)
+            vf2_call_limit = None
+            if (
+                pass_manager_config.layout_method is None
+                and pass_manager_config.initial_layout is None
+            ):
+                vf2_call_limit = int(5e6)  # Set call limit to ~10 sec with retworkx 0.10.2
```

Comment:
A really minor comment. The lines:
```
vf2_call_limit = None
if (
    pass_manager_config.layout_method is None
    and pass_manager_config.initial_layout is None
):
    vf2_call_limit = int(5e6)
```
are repeated many times in this file (with the value of `vf2_call_limit` depending on `opt_level`). Would it make sense to wrap this in some auxiliary function, something like `vf2_call_limit = _get_vf2_call_limit(pass_manager_config, opt_level)`?

Reply:
Yeah, that's a good idea. There is probably also a bunch of other deduplication we can do in the built in plug-ins i made most of it by copy and paste.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/8305/commits/5dec6c81e7afaba95b0bf1ee150a297030002a26